package.json

```diff
@@ -26,6 +26,7 @@
     "server": "node src/backend/web/server",
     "test-ci": "run-s prettier-check test",
     "pre-commit": "pretty-quick --staged",
+    "preversion": "node tools/collect-dependencies.js -- -o src/api/dependency-discovery/deps.txt && git add src/api/dependency-discovery/deps.txt",
```

Comment:
Question: will this get committed by `version`?  I assume so, but haven't tested.

Reply:
We better add some docs to the `README.md` in your dependency service to explain this logic and how/where this file comes from.

Reply:
> Question: will this get committed by `version`? I assume so, but haven't tested.

I wanted to check whether that is true, so I head to the [documentation of `npm version`](https://docs.npmjs.com/cli/v8/commands/npm-version#description) to see if this was mentioned.

It seems that it will committed as long as you add added them to staging with `git add` before the commit done by `npm version`.